ui/core/src/api/MarketService/index.ts

```diff
@@ -1,80 +1,67 @@
-import { RWN } from "../../constants";
-import { Asset, AssetAmount, Pool } from "../../entities";
-import { Fraction } from "../../entities/fraction/Fraction";
+import ReconnectingWebSocket from "reconnecting-websocket";
+import { Asset, Pool } from "../../entities";
 import { SifUnSignedClient } from "../utils/SifClient";
-import { RawPool } from "../utils/x/clp";
+import { toPool } from "../utils/toPool";
 
 export type MarketServiceContext = {
   loadAssets: () => Promise<Asset[]>;
   sifApiUrl: string;
   nativeAsset: Asset;
 };
 
-function toAssetSymbol(assetOrString: Asset | string) {
-  return typeof assetOrString === "string"
-    ? assetOrString
-    : assetOrString.symbol;
-}
-
-function makeQuerablePromise<T>(promise: Promise<T>) {
-  let isResolved = false;
-
-  promise.then(() => {
-    isResolved = true;
-  });
-
-  return {
-    isResolved() {
-      return isResolved;
-    },
-  };
-}
-
-function processPool(poolData: RawPool) {
-  const externalAssetTicker = poolData.external_asset.ticker;
-
-  return Pool(
-    AssetAmount(RWN, poolData.native_asset_balance),
-    AssetAmount(
-      Asset.get(externalAssetTicker),
-      poolData.external_asset_balance
-    ),
-    new Fraction(poolData.pool_units)
-  );
-}
+type PoolHandlerFn = (pools: Pool[]) => void;
 
 export default function createMarketService({
   loadAssets,
   sifApiUrl,
 }: MarketServiceContext) {
+  let ws: ReconnectingWebSocket;
   const sifClient = new SifUnSignedClient(sifApiUrl);
-  const poolMap = new Map<string, Pool>();
+
+  let poolHandler: PoolHandlerFn = () => {};
+
+  async function setupPoolWatcher() {
+    await new Promise((res, rej) => {
+      ws = new ReconnectingWebSocket("ws://localhost:26657/websocket");
```

Comment:
This is good. Will need to tie this into a chain config file (which we'll need for keplr wallet too)

Reply:
I'd put a TODO here or create a chain config like this: https://github.com/chainapsis/keplr-example/blob/master/src/main.js#L19

Reply:
Yep we definitely need to sort out config including a coin whitelist